nucypher/policy/conditions/lingo.py

```diff
@@ -104,21 +106,44 @@ def __init__(self, comparator: str, value: Any):
 
         self.comparator = comparator
         self.value = value
+        self.key = key
 
     def _sanitize_value(self, value):
         try:
             return ast.literal_eval(str(value))
         except Exception:
             raise self.InvalidExpression(f'"{value}" is not a permitted value.')
 
+    def _process_data(self, data: Any) -> Any:
+        """
+        If a key is specified, return the value at that key in the data if data is a dict or list-like.
+        Otherwise, return the data.
+        """
+        if self.key and isinstance(data, dict):
+            try:
+                processed_data = data[self.key]
+            except KeyError:
+                raise KeyError(f"Key '{self.key}' not found in return data.")
+        elif isinstance(self.key, int) and isinstance(data, (list, tuple)):
+            try:
+                processed_data = data[self.key]
+            except IndexError:
+                raise IndexError(f"Index '{self.key}' not found in return data.")
+        else:
+            processed_data = data
+
+        return processed_data
+
     def eval(self, data) -> bool:
-        if is_context_variable(self.value):
+        if is_context_variable(self.value) or is_context_variable(self.key):
             # programming error if we get here
             raise RuntimeError(
-                f"'{self.value}' is an unprocessed context variable and is not valid "
+                f"Either '{self.value}' or '{self.key}' is an unprocessed context variable and is not valid "
                 f"for condition evaluation."
```

Comment:
What about something more concrete like:
```suggestion
                f"Return value comparator contains an unprocessed context variable (key={self.key}, value={self.value}) and is not valid "
                f"for condition evaluation."
```